src/components/CustomAppBar.js

```diff
@@ -0,0 +1,94 @@
+// Custom App Bar with a "Recipes" title and clickable SearchIcon,
+// which is responsible for showing Search Bar. The component has
+// been implemented on the basis of: https://material-ui.com/components/app-bar/
+
+
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import IconButton from '@material-ui/core/IconButton';
+import SearchIcon from '@material-ui/icons/Search';
+import CloseIcon from '@material-ui/icons/Close'
+
+
```

Comment:
Couple things to correct:
- Add callback when the user is submitting the search
- Make app bar header not selectable by user